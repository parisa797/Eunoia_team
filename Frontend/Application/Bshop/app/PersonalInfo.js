import React, { useEffect, useState } from "react";
import {
  ScrollView,
  Button,
  TextInput,
  Text,
  Image,
  View,
  Alert,
  StyleSheet,
  ToastAndroid,
  TouchableOpacity,
  Container,
} from "react-native";
import axios from "axios";
import * as SecureStore from "expo-secure-store";

const PersonalInfo = ({ navigation }) => {
  const [info, setinfo] = useState("");

  const [name, setname] = useState();
  const [familyname, setfamilyname] = useState();
  const [phone, setphone] = useState();
  const [address, setaddress] = useState();

  const phoneLenght = (text) => {
    console.log("text is", text, "09125831292");
    let pattern = /^[0][1-9]\d{9}$|^[1-9]\d{9}$/;
    return pattern.test(String(text));
  };

  //get user's info
  // useEffect(async () => {
  //   // (async () => {
  //   var myHeaders = new Headers();
  //   let result = await SecureStore.getItemAsync("token");
  //   console.log("result");
  //   var SaveInfo = "Token " + result;
  //   // var SaveInfo ="Token " + getValueFor("token")
  //   console.log("token is:");
  //   console.log(SaveInfo);
  //   myHeaders.append("Authorization", SaveInfo);

  //   var requestOptions = {
  //     method: "GET",
  //     headers: myHeaders,
  //     redirect: "follow",
  //   };
  //   getValueFor("token")
  //   fetch("http://iust-bshop.herokuapp.com/users/profile", requestOptions)
  //     .then((response) => response.json())
  //     .then((result) => {
  //       setinfo(result);
  //       setname(result.FirstName);
  //       setfamilyname(result.LastName);
  //       setphone(result.phone);
  //       setaddress(result.address);
  //       console.log(result);
  //     })
  //     .catch((error) => console.log("error", error));
  //   })();
  // }, []);

  useEffect(() => {
    const getResult = async () => {
      var myHeaders = new Headers();
      let result = await SecureStore.getItemAsync("token");
      console.log("result");
      var SaveInfo = "Token " + result;
      // var SaveInfo ="Token " + getValueFor("token")
      console.log("token is:");
      console.log(SaveInfo);
      myHeaders.append("Authorization", SaveInfo);

      var requestOptions = {
        method: "GET",
        headers: myHeaders,
        redirect: "follow",
      };
      fetch("http://eunoia-bshop.ir:8000/users/profile", requestOptions)
        .then((response) => response.json())
        .then((result) => {
          console.log("hey");
          setinfo(result);

          setname(result.FirstName);

          setfamilyname(result.LastName);

          setphone(result.phone);

          setaddress(result.address);

          console.log(result);
        })
        .catch((error) => console.log("error", error));
    };

    getResult();
  }, []);

  //update user's info when pressing update button
  async function beruzresani() {
    console.log("beruzresani");
    var myHeaders = new Headers();
    let result = await SecureStore.getItemAsync("token");
    var SaveInfo = "Token " + result;
    console.log("result is" + result);
    myHeaders.append("Authorization", SaveInfo);
    myHeaders.append("Content-Type", "multipart/form-data");
    var formdata = new FormData();
    formdata.append("email", info.email);
    formdata.append("FirstName", name);
    formdata.append("LastName", familyname);
    formdata.append("address", address);
    formdata.append("phone", phone);
    formdata.append("files", []);
    console.log(formdata);
    console.log("formdata");
    var requestOptions = {
      method: "PUT",
      headers: myHeaders,
      body: formdata,
    };

    //   fetch("http://eunoia-bshop.ir:8000/users/profile", requestOptions)
    //     .then((response) => {
    //       console.log("parisaaa");
    //       if (response.status == 200) {
    //         alert("heeeeeeelo");
    //         return response.json();
    //       }
    //     })
    //     .then((result) => console.log(result))
    //     .catch((error) => console.log("error", error));
    // }

    const response = await axios
      .put("http://eunoia-bshop.ir:8000/users/profile", formdata, {
        headers: {
          "Content-Type":
            "multipart/form-data; boundary=<calculated when request is sent>",
          Authorization:
            "Token " + (await SecureStore.getItemAsync("token")).toString(),
        },
      })
      .then(function (response) {
        console.log("setayesh");
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  return (
    <ScrollView nestedScrollEnabled={true} style={styles.kooft}>
      <View style={styles.container}>
        {/* <Image style={styles.image} source={require("../assets/profile.png")} /> */}
        <View style={styles.inputView}>
          {/* <View style={styles.bazgasht}>
            <TouchableOpacity
              style={styles.Btn}
              onPress={navigation.navigate("Home")}
            >
              <Text>بازگشت </Text>
            </TouchableOpacity>
          </View> */}
          <TextInput
            style={styles.input}
            value={name}
            icon="user"
            placeholder="نام"
            autoCapitalize="none"
            autoCompleteType="name"
            textAlign="center"
            testID={"namepersonal_check"}
            onChangeText={setname}
          />
        </View>
        <View style={styles.inputView}>
          <TextInput
            style={styles.input}
            icon="user"
            placeholder="نام خانوادگی"
            autoCapitalize="none"
            // autoCompleteType="familyname"
            textAlign="center"
            value={familyname}
            testID={"family_check"}
            onChangeText={setfamilyname}
          />
        </View>
        <View style={styles.inputView}>
          <TextInput
            style={styles.input}
            icon="phone"
            placeholder="شماره تلفن"
            // autoCompleteType="phone"
            autoCapitalize="none"
            textAlign="center"
            value={phone}
            testID={"phone_check"}
            onChangeText={setphone}
            onBlur={() => {
              if (!phoneLenght(phone))
                ToastAndroid.show(
                  "لطفا شماره تلفن همراه خود را به صورت کامل وارد کنید. الگو : 091*******25",
                  ToastAndroid.SHORT
                );
            }}
          />
        </View>
        <View style={styles.inputView}>
          <TextInput
            style={styles.input}
            icon="map-pin"
            placeholder="آدرس"
            autoCapitalize="none"
            textAlign="center"
            value={address}
            testID={"address_check"}
            onChangeText={setaddress}
            // autoCompleteType="street-address"
          />
        </View>
        <View style={styles.inputView}>
          <TextInput
            style={styles.input}
            // icon="email"
            placeholder="ایمیل"
            // autoCompleteType="email"
            autoCapitalize="none"
            textAlign="center"
            value={info.email}
            readonly={true}
          />
        </View>
        <View style={styles.kenareham}>
          <TouchableOpacity style={styles.Btn} onPress={() => beruzresani()}>
            <Text>ذخیره اطلاعات </Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.Btn}>
            <Text>حذف حساب کاربری </Text>
          </TouchableOpacity>
        </View>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  input: {
    width: 300,
    height: 55,
    backgroundColor: "#f1f1f2",
    marginBottom: 20,
    padding: 8,
    color: "black",
    borderRadius: 14,
    fontSize: 18,
    fontWeight: "500",
    textAlign: "center",
  },
  kooft: {
    flex: 1,
    backgroundColor: "#fff",
    // alignItems: "center",
    // justifyContent: "center",
    marginTop: "5%",
    // width: "50%",
    // height: "50%",
    marginBottom: "5%",
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#fff",
    marginTop: "40%",
  },
  Btn: {
    width: 130,
    borderRadius: 10,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 20,
    marginBottom: 5,
    backgroundColor: "#b31414",
  },
  bazgasht: {
    width: 130,
    borderRadius: 10,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 20,
    marginBottom: 5,
    backgroundColor: "#b31414",
  },
  kenareham: {
    justifyContent: "space-between",
    flexDirection: "row-reverse",
  },
  image: {
    width: "50%",
    height: "20%",
    marginTop: -35,
  },
});
export default PersonalInfo;
