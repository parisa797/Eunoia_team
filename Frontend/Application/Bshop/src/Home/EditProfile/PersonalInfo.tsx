import React, { useEffect, useState } from "react";
import { ScrollView , Button } from "react-native";
import * as SecureStore from "expo-secure-store";
import { Box, Text } from "../../components";
import TextInput from "../../components/Form/TextInput";
import CheckboxGroup from "./CheckboxGroup";


async function getValueFor(key) {
  let result = await SecureStore.getItemAsync(key);
  // if (result) {
  //   console.log(" Here's your value  \n" + result);
  // } else {
  //   console.log('No values stored under that key.');
  // }
  // inja
  console.log("hello")
  console.log(result)
  return(result)
}

const PersonalInfo = () => {
// const Token = getValueFor("token")
// console.log("i in personal")
// console.log(Token)
const update=async() =>{
  var myHeaders = new Headers();
  let result = await SecureStore.getItemAsync("token");
  var SaveInfo ="Token " + result
  console.log("result is" + result)
myHeaders.append("Authorization", SaveInfo);

var formdata = new FormData();
formdata.append("email",info.email);
formdata.append("FirstName",name);
formdata.append("LastName",familyname);
formdata.append("address",address);
formdata.append("phone",phone);
formdata.append("files",[]);

var requestOptions = {
  method: 'PUT',
  headers: myHeaders,
  body: formdata,
  redirect: 'follow'
};

fetch("http://10.0.2.2:8000/users/profile", requestOptions)
  .then(response => response.text())
  .then(result => console.log(result))
  .catch(error => console.log('error', error));

}
  
const [info ,setinfo]=useState("")


const [name ,setname]=useState()
const [familyname ,setfamilyname]=useState()
const [phone ,setphone]=useState()
const [address ,setaddress]=useState()
// const getinfo=()=>{
  useEffect(async() => {
//  async function fetchdata() {
   
  var myHeaders = new Headers();
  let result = await SecureStore.getItemAsync("token");
  var SaveInfo ="Token " + result
  // var SaveInfo ="Token " + getValueFor("token")
  // console.log(SaveInfo)
myHeaders.append("Authorization", SaveInfo);

var requestOptions = {
  method: 'GET',
  headers: myHeaders,
  redirect: 'follow'
};
// getValueFor("token")
fetch("http://10.0.2.2:8000/users/profile", requestOptions)
  .then(response => response.json())
  .then((result) => {
  setinfo(result)
  setname(result.FirstName)
  setfamilyname(result.LastName)
  setphone(result.phone)
  setaddress(result.address)
        console.log(result);
      })
  .catch(error => console.log('error', error));
//  }
//  fetchdata()
},[])

// }
  return (
    <ScrollView>
      <Box padding="m">
        <Text variant="body" marginBottom="m">اطلاعات حساب</Text>
        <Box marginBottom="m">
          <TextInput
            icon="user"
            placeholder="نام"
            autoCapitalize="none"
            autoCompleteType="name"
             textAlign="center"
             value={name}
             onChangeText={setname}
          />
        </Box>
         <Box marginBottom="m">
          <TextInput
            icon="user"
            placeholder="نام خانوادگی"
            autoCapitalize="none"
            // autoCompleteType="familyname"
             textAlign="center"
             value={familyname}
             onChangeText={setfamilyname}
          />
        </Box>
        <Box marginBottom="m">
          <TextInput
            icon="phone"
            placeholder="شماره تلفن"
            // autoCompleteType="phone"
            autoCapitalize="none"
             textAlign="center"
             value={phone}
             onChangeText={setphone}
            
          />
        </Box>
        <Box marginBottom="m">
          <TextInput
            icon="map-pin"
            placeholder="آدرس"
            autoCapitalize="none"
            textAlign="center"
            value={address}
            onChangeText={setaddress}
            // autoCompleteType="street-address"
          />
        </Box>
          <Box marginBottom="m">
          <TextInput
            // icon="email"
            placeholder="ایمیل"
            // autoCompleteType="email"
            autoCapitalize="none"
             textAlign="center"
             value={info.email}
             readonly={true}
            
          />
        </Box>
        <Button 
          title="ذخیره اطلاعات  " 
          onPress={() => update()}
        />
        </Box>
    </ScrollView>
  );
};

export default PersonalInfo;
