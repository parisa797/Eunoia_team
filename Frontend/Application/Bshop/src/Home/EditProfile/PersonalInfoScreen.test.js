import React from "react";
import renderer from "react-test-renderer";
import { act } from "react-test-renderer";
import PersonalInfo from "./PersonalInfo";
// import { getByTestId } from "@testing-library/dom";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
} from "@testing-library/react-native";
import ReactDOM from "react-dom";
import { Alert } from "react-native";
// import "jest-dom/extend-expect";

global.fetch = jest.fn();

// mocking an API success response once
fetch.mockResponseIsSuccess = (body) => {
  fetch.mockImplementationForOnce(() =>
    Promise.resolve({ json: () => Promise.resolve(JSON.parse(body)) })
  );
};

// mocking an API failure response for once
fetch.mockResponseIsFailure = (error) => {
  fetch.mockImplementationForOnce(() => Promise.reject(error));
};

describe("PersonalInfo component be tested", () => {
  // afterEach(cleanup);

  it("renders correctly", () => {
    const tree = renderer.create(<PersonalInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //   it("check email input", async () => {
  //     const push = jest.fn();
  //     const { getByTestId, getByPlaceholderText } = await render(
  //       <PersonalInfo navigation={{ push }} />
  //     );
  //     const eml = "hi@gmail.com";
  //     await fireEvent.changeText(getByTestId("eml_check"), eml);
  //     expect(getByTestId("eml_check").props.value).toBe(eml);
  //   });

  it("check namepersonal input", async () => {
    const push = jest.fn();
    const { getByTestId, getByPlaceholderText } = await render(
      <PersonalInfo navigation={{ push }} />
    );
    const name = "hojat";
    await fireEvent.changeText(getByTestId("namepersonal_check"), name);
    expect(getByTestId("namepersonal_check").props.value).toBe(name);
  });

  it("check family_check input", async () => {
    const push = jest.fn();
    const { getByTestId, getByPlaceholderText } = await render(
      <PersonalInfo navigation={{ push }} />
    );
    const familyname = "amini";
    await fireEvent.changeText(getByTestId("family_check"), familyname);
    expect(getByTestId("family_check").props.value).toBe(familyname);
  });

  it("check phone_check input", async () => {
    const push = jest.fn();
    const { getByTestId, getByPlaceholderText } = await render(
      <PersonalInfo navigation={{ push }} />
    );
    const phone = "09125831292";
    await fireEvent.changeText(getByTestId("phone_check"), phone);
    expect(getByTestId("phone_check").props.value).toBe(phone);
  });

  it("check address_check input", async () => {
    const push = jest.fn();
    const { getByTestId, getByPlaceholderText } = await render(
      <PersonalInfo navigation={{ push }} />
    );
    const address = "mirdamad";
    await fireEvent.changeText(getByTestId("address_check"), address);
    expect(getByTestId("address_check").props.value).toBe(address);
  });

  //   it("check email validation", async () => {
  //     const push = jest.fn();
  //     const { getByTestId, getByPlaceholderText } = await render(
  //       <SignUp navigation={{ push }} />
  //     );

  //     errorMessageText = "لطفا ایمیل خود را درست وارد کنید.";
  //     Alert.alert = jest.fn().mockResolvedValue(errorMessageText);
  //     // jest.spyOn(Alert, "alert");

  //     const emailvalidation = "phj.com";
  //     await fireEvent.changeText(getByTestId("eml_check"), emailvalidation);
  //     // expect(getByTestId("eml_check").props.value).toBe(emailvalidation);
  //     // expect(Alert.alert).toHaveBeenCalledWith(errorMessageText);
  //   });

  //   it("check same pass and confpass", async () => {
  //     const push = jest.fn();
  //     const { getByTestId, getByPlaceholderText } = await render(
  //       <SignUp navigation={{ push }} />
  //     );

  //     errorMessageText = "رمزعبورهای وارد شده باید با هم مطابقت داشته باشند.";
  //     Alert.alert = jest.fn().mockResolvedValue(errorMessageText);
  //     // jest.spyOn(Alert, "alert");
  //     const pass = "13799731p";
  //     await fireEvent.changeText(getByTestId("pass_check"), pass);
  //     const samepass = "13799731pp";
  //     await fireEvent.changeText(getByTestId("confpass_check"), samepass);
  //     // expect(getByTestId("eml_check").props.value).toBe(emailvalidation);
  //     // expect(Alert.alert).toHaveBeenCalledWith(errorMessageText);
  //   });

  //   it("check username has number and alphabet", async () => {
  //     const push = jest.fn();
  //     const { getByTestId, getByPlaceholderText } = await render(
  //       <SignUp navigation={{ push }} />
  //     );

  //     errorMessageText =
  //       "نام کاربری نمیتواند شامل ارقام باشد! لطفا از ترکیبی از حرف و عدد استفاده کنید.";
  //     Alert.alert = jest.fn().mockResolvedValue(errorMessageText);
  //     // jest.spyOn(Alert, "alert");

  //     const cheakboathuserform = "1254";
  //     await fireEvent.changeText(getByTestId("user_cheak"), cheakboathuserform);
  //     // expect(getByTestId("eml_check").props.value).toBe(emailvalidation);
  //     // expect(Alert.alert).toHaveBeenCalledWith(errorMessageText);
  //   });
});
