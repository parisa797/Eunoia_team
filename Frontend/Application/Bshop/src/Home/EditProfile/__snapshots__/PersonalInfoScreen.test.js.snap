// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfo component be tested renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginBottom": "5%",
      "marginTop": "5%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": "40%",
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="name"
          icon="user"
          onChangeText={[Function]}
          placeholder="نام"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#f1f1f2",
              "borderRadius": 14,
              "color": "black",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "marginBottom": 20,
              "padding": 8,
              "textAlign": "center",
              "width": 300,
            }
          }
          testID="namepersonal_check"
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          icon="user"
          onChangeText={[Function]}
          placeholder="نام خانوادگی"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#f1f1f2",
              "borderRadius": 14,
              "color": "black",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "marginBottom": 20,
              "padding": 8,
              "textAlign": "center",
              "width": 300,
            }
          }
          testID="family_check"
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          icon="phone"
          onChangeText={[Function]}
          placeholder="شماره تلفن"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#f1f1f2",
              "borderRadius": 14,
              "color": "black",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "marginBottom": 20,
              "padding": 8,
              "textAlign": "center",
              "width": 300,
            }
          }
          testID="phone_check"
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          icon="map-pin"
          onChangeText={[Function]}
          placeholder="آدرس"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#f1f1f2",
              "borderRadius": 14,
              "color": "black",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "marginBottom": 20,
              "padding": 8,
              "textAlign": "center",
              "width": 300,
            }
          }
          testID="address_check"
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          placeholder="ایمیل"
          readonly={true}
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#f1f1f2",
              "borderRadius": 14,
              "color": "black",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "marginBottom": 20,
              "padding": 8,
              "textAlign": "center",
              "width": 300,
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#b31414",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 20,
              "opacity": 1,
              "width": 130,
            }
          }
        >
          <Text>
            ذخیره اطلاعات 
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#b31414",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginTop": 20,
              "opacity": 1,
              "width": 130,
            }
          }
        >
          <Text>
            حذف حساب کاربری 
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
